test/tools/project-cleaner/main.go

```diff
@@ -112,6 +114,69 @@ func (p *ProjectCleaner) GKEClusterHandler(resource GCPResource) {
 	}
 }
 
+func (p *ProjectCleaner) PersistentDiskHandler(resource GCPResource) {
+	// See https://cloud.google.com/docs/authentication/.
+	// Use GOOGLE_APPLICATION_CREDENTIALS environment variable to specify a service account key file
+	// to authenticate to the API.
+	ctx := context.Background()
+	c, err := compute.NewDisksRESTClient(ctx)
+	if err != nil {
+		log.Fatalf("Could not initialize perisistent disk client: %v", err)
+	}
+	defer c.Close()
+
+	for _, zone := range resource.Zones {
+		// maxResults := uint32(2)
+		// order := "creationTimestamp desc"
+		req := &computepb.ListDisksRequest{
+			Project: p.ProjectId,
+			Zone:    zone,
+			// OrderBy:    &order,
+			// MaxResults: &maxResults,
+		}
+		disk_list, listerr := c.List(ctx, req)
+		if listerr != nil {
+			log.Fatalf("Could not get perisistent disk list: %v, zone: %s", listerr, zone)
+		}
+		diskmarshal, _ := json.Marshal(disk_list)
+		log.Printf("disk_list:%s", string(diskmarshal))
+		log.Printf("disk_list size:%d", len(disk_list.GetItems()))
+
+		for _, disk := range disk_list.GetItems() {
+
+			creationTimestamp := disk.GetCreationTimestamp()
+			createdTime, _ := time.Parse(time.RFC3339, creationTimestamp)
+			duration := time.Since(createdTime)
+
+			// If the disk is not detached yet, do not delete the persistent disk.
+			if len(disk.GetLastDetachTimestamp()) == 0 {
+				log.Printf("disk: %s doesn't have last detach time, do not delete.", disk.GetName())
+				continue
+			}
+			log.Printf("disk: %s last detach time: %s", disk.GetName(), disk.GetLastDetachTimestamp())
+
+			// Do not delete the persistent disk if this is less than 30 days old.
+			if duration.Hours() < 24*30 {
```

Comment:
We don't need to keep disks for such a long time, probably 2 hours is enough, we only need to make sure, it doesn't hurt tests.

Reply:
Because we are comparing creation time vs current, I am guessing it might be helpful to be slightly longer than 2 hours? Tentatively setting it to 1 day, and we can change it later if not enough.

Reply:
Because we already filter disks that have been detached, shorter time feels enough to me.
Because if we set it to 1 day, we'll be charging for all the disks for 1 day, but we only need them for 1 hour running the test.